packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -295,13 +295,18 @@ export abstract class BaseFetchService<DS extends BaseDataSource, B extends IBlo
     scaledBatchSize: number,
     latestHeight: number
   ): Promise<void> {
-    const estRangeEndHeight = this.nextEndBlockHeight(startBlockHeight, scaledBatchSize);
+    // End height from current dataSource
     const {endHeight, value: relevantDs} = this.getRelevantDsDetails(startBlockHeight);
+    // Estimated range end height
+    const estRangeEndHeight = Math.min(
+      endHeight ?? Number.MAX_SAFE_INTEGER,
+      this.nextEndBlockHeight(startBlockHeight, scaledBatchSize)
+    );
```

Comment:
Does this work? Then we can get rid of the other math.min

```suggestion
    const estRangeEndHeight = Math.min(
      endHeight ?? Number.MAX_SAFE_INTEGER,
      this.nextEndBlockHeight(startBlockHeight, scaledBatchSize),
      latestHeight
    );
```